enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -93,17 +102,37 @@ func (c *podmanCommandContainer) Run(ctx context.Context, command *repb.Command,
 }
 
 func (c *podmanCommandContainer) Create(ctx context.Context, workDir string) error {
-	c.workDir = workDir
-	return nil
+	containerName, err := generateContainerName()
+	if err != nil {
+		return status.UnavailableErrorf("failed to generate podman container name: %s", err)
+	}
+	c.name = containerName
+
+	podmanRunArgs := c.getPodmanRunArgs(workDir, false /*=autoRemove*/)
+	podmanRunArgs = append(podmanRunArgs, c.image)
+	podmanRunArgs = append(podmanRunArgs, "sleep", "infinity")
+	createResult := runPodman(ctx, "create", nil, nil, podmanRunArgs...)
+	if err = createResult.Error; err != nil {
+		return status.UnavailableErrorf("failed to create container: %s", err)
+	}
+
+	startResult := runPodman(ctx, "start", nil, nil, c.name)
+	return startResult.Error
 }
 
 func (c *podmanCommandContainer) Exec(ctx context.Context, cmd *repb.Command, stdin io.Reader, stdout io.Writer) *interfaces.CommandResult {
-	return runPodman(ctx, "run" /*workDir=*/, "", stdin, stdout, cmd.Arguments...)
+	podmanRunArgs := []string{}
```

Comment:
you can make this the right capacity ahead of time and avoid extra allocation work:

podmanRunArgs := make([]string, 0, len(cmd.GetEnvironmentVariables()))

Reply:
Done